SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,230 @@
+      DataT *shared_init_values_ptr = shared_init_values.data();
+      DataT *result_ptr = shared_result.data();
+      size_t *const indexes_to_change_ptr = shared_indexes_to_change.data();
+
+      cgh.parallel_for<
+          Kernel<DataT, NumElems, TestCaseT, ChosenOperatorT, NDRangeDimT,
+                 NumElemsToChangeT, ChangeElemFilterT, AlgorithmToChangeT>>(
+          nd_range, [=](sycl::nd_item<1> nd_item) SYCL_ESIMD_KERNEL {
+            const size_t work_item_index = nd_item.get_global_linear_id();
+            TestCaseT::template simd_function<DataT, NumElems, NumElemsToChange,
+                                              ChangeElemFilterT,
+                                              ChosenOperator>(
+                shared_init_values_ptr, result_ptr, indexes_to_change_ptr,
+                work_item_index);
+          });
+    });
+    queue.wait_and_throw();
+
+    std::sort(shared_result.begin(), shared_result.end());
+
+    ChangeElemFilterT filter{};
+    size_t num_changed_elems = 0;
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        ++num_changed_elems;
+      }
+    }
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      const DataT &retrieved = shared_result[i];
+      if (i < NumberIteractions) {
+        if (result_fill_value != retrieved) {
+          passed = fail_test(i, result_fill_value, retrieved, data_type,
+                             "atomic_update return value");
+        }
+      } else {
+        const DataT &expected = base_value * num_changed_elems +
+                                (i - NumberIteractions) * num_changed_elems;
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "atomic_update return value");
+        }
+      }
+    }
+
+    std::vector<size_t> selected_indexes;
+    // Collect the indexess that has been selected.
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        selected_indexes.push_back(indexes_to_change[i]);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto next_selected_index = selected_indexes.begin();
+
+    const DataT &expected = base_value;
+    DataT expected_after_change = functions::get_expected_value<ChosenOperator>(
+        base_value, NumberIteractions);
+    // Verify that values, that do not was changed has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been changed.
+      const DataT &retrieved = shared_init_values[i];
+      if (i < *next_selected_index) {
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "value that should be updated");
```

Comment:
Maybe
```suggestion
                             "value that should not be updated");
```
Did I get it right?

Reply:
I've applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/939/commits/8f6071023226997f11c6ff346027fdfcf14b5c24